airthings_ble/device_type.py

```diff
@@ -0,0 +1,83 @@
+"""Airthings device types."""
+
+from enum import Enum
+
+
+class AirthingsDeviceType(Enum):
+    """Airthings device types."""
+
+    UNKNOWN = 0
+    WAVE_GEN_1 = "2900"
+    WAVE_MINI = "2920"
+    WAVE_PLUS = "2930"
+    WAVE_RADON = "2950"
+
+    raw_value: str  # pylint: disable=invalid-name
+
+    def __new__(cls, value: str) -> "AirthingsDeviceType":
+        """Create new device type."""
+        obj = object.__new__(cls)
+        obj.raw_value = value
+        return obj
+
+    @classmethod
+    def from_raw_value(cls, value: str) -> "AirthingsDeviceType":
+        """Get device type from raw value."""
+        for device_type in cls:
+            if device_type.value == value:
+                device_type.raw_value = value
+                return device_type
+        unknown_device = AirthingsDeviceType.UNKNOWN
+        unknown_device.raw_value = value
+        return unknown_device
+
+    @property
+    def product_name(self) -> str:
+        """Get product name."""
+        if self == AirthingsDeviceType.WAVE_GEN_1:
+            return "Wave Gen 1"
+        if self == AirthingsDeviceType.WAVE_MINI:
+            return "Wave Mini"
+        if self == AirthingsDeviceType.WAVE_PLUS:
+            return "Wave Plus"
+        if self == AirthingsDeviceType.WAVE_RADON:
+            return "Wave Radon"
+        return "Unknown"
+
+    def battery_percentage(self, vin: float) -> int:
+        """Calculate battery percentage based on voltage."""
+        if self == AirthingsDeviceType.WAVE_MINI:
+            return self._battery_percentage_wave_mini(vin)
+        return self._battery_percentage_wave_radon_and_plus(vin)
+
+    # pylint: disable=too-many-return-statements
+    def _battery_percentage_wave_radon_and_plus(self, vin: float) -> int:
+        if vin >= 3.00:
```

Comment:
We can round the values to 2 decimals. However, we must know [different round behaviors in Python](https://stackoverflow.com/questions/10825926/python-3-x-rounding-behavior). 

It is possibly solvable by a contextual round implementation.

```python
def round_voltage(x: float, n: int) -> float:
    return round(x + 0.00001, n)  # Avoid rounding 2.995 to 2.99, etc.
```